src/tblite_interface.F

```diff
@@ -0,0 +1,1790 @@
+!> \brief ...
+!> \param qs_env ...
+!> \param tb ...
+! **************************************************************************************************
+   SUBROUTINE tb_init_geometry(qs_env, tb)
+
+      TYPE(qs_environment_type), POINTER                 :: qs_env
+      TYPE(tblite_type), POINTER                         :: tb
+      
+#if defined(__TBLITE)
+
+      CHARACTER(LEN=*), PARAMETER :: routineN = 'tblite_init_geometry'
+
+      TYPE(cell_type), POINTER                           :: cell
+      TYPE(particle_type), DIMENSION(:), POINTER         :: particle_set
+      TYPE(qs_kind_type), DIMENSION(:), POINTER          :: qs_kind_set
+      INTEGER                                            :: iatom, natom
+      REAL(KIND=dp), ALLOCATABLE, DIMENSION(:, :)        :: xyz
+      INTEGER                                            :: handle, ikind
+      INTEGER, DIMENSION(3)                              :: periodic
+      LOGICAL, DIMENSION(3)                              :: lperiod
+
+      CALL timeset(routineN, handle)
+
+      !get info from environment vaiarable
+      CALL get_qs_env(qs_env=qs_env, particle_set=particle_set, cell=cell, qs_kind_set=qs_kind_set)
+
+      !get information about particles
+      natom = SIZE(particle_set)
+      ALLOCATE (xyz(3, natom))
+      CALL allocate_tblite_type(tb)
+      ALLOCATE (tb%el_num(natom))
+      tb%el_num = -9
+      DO iatom = 1, natom
+         xyz(:, iatom) = particle_set(iatom)%r(:)
+         CALL get_atomic_kind(particle_set(iatom)%atomic_kind, kind_number=ikind)
+         CALL get_qs_kind(qs_kind_set(ikind), zatom=tb%el_num(iatom))
+         IF (tb%el_num(iatom) < 1 .OR. tb%el_num(iatom) > 85) THEN
+            CPABORT("only elements 1-85 are supported by tblite")
+         END IF
+      END DO
+
+      !get information about cell / lattice
+      CALL get_cell(cell=cell, periodic=periodic)
+      lperiod(1) = periodic(1) == 1
+      lperiod(2) = periodic(2) == 1
+      lperiod(3) = periodic(3) == 1
+
+      !prepare for the call to the dispersion function
+      CALL new(tb%mol, tb%el_num, xyz, lattice=cell%hmat, periodic=lperiod)
+
+      DEALLOCATE (xyz)
+
+      CALL timestop(handle)
+
+#else
+      CPABORT("Build without TBLITE")
+#endif
+
+   END SUBROUTINE tb_init_geometry
+
+   ! **************************************************************************************************
+!> \brief ...
+!> \param tb ...
+!> \param do_grad ...
+! **************************************************************************************************
+   SUBROUTINE tb_init_wf(tb, do_grad)
+
+      TYPE(tblite_type), POINTER                         :: tb
+      LOGICAL, INTENT(in)                                :: do_grad
+      
+#if defined(__TBLITE)
+
+      REAL(KIND=dp), PARAMETER                           :: K2Eh = 3.16681156346E-06
+      INTEGER                                            :: nSpin = 1 !number of spin channels
```

Comment:
Is this supposed to be a constant? If yes, add PARAMETER, otherwise, make use of `SAVE`. All non-`PARAMETER` variables implicitly have the `SAVE` attribute in Fortran (like static variables).